packages/stark-ui/src/modules/route-search/components/route-search.component.ts

```diff
@@ -0,0 +1,199 @@
+import {Component, Inject, Input, OnInit, ViewEncapsulation} from "@angular/core";
+import { RouteEntry } from "../entities/route-entry";
+import { Observable } from "rxjs/internal/Observable";
+import { FormControl } from "@angular/forms";
+import {map, startWith} from "rxjs/operators";
+import { STARK_ROUTING_SERVICE, StarkRoutingService, STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import { UIRouter } from "@uirouter/core";
+import {StarkMenuConfig, StarkMenuGroup, StarkMenuSection} from "@nationalbankbelgium/stark-ui";
+export type StarkRouteSearchDirection = "left" | "right" | "";
+
+
+/**
+ * Name of the component
+ *
+ */
+const componentName: string = "route-search";
+
+/**
+ * Component to search a particular route with autocompletion that will redirect to the route chosen by the user
+ */
+@Component({
+	selector: "route-search",
+	templateUrl: "./route-search.component.html",
+	encapsulation: ViewEncapsulation.None,
+	host: {
+		class: componentName
+	}
+})
+export class StarkRouteSearchComponent implements OnInit {
+	
+	/**
+	 * the controller of the form
+	 */
+	public pathCtrl: FormControl;
+
+	/**
+	 * the list of routes filtered by autocomplete
+	 */
+	public filteredOptions: Observable<any[]>;
+
+	/**
+	 * the list of routes filtered by autocomplete
+	 */
+	public optionsToDisplay: RouteEntry[] = [];
+	
+	/**
+	 * The menu which options will be displayed
+	 */
+	@Input()
+	public optionsList?: StarkMenuConfig;
```

Comment:
Can you rename this variable to `menuConfig` so it's clear what content is expected to be there?